### PR TITLE
Added support for string substitution in XML files

### PR DESCRIPTION
**Pull request by brian-learningpool**
This is required to support upcoming work on the xAPI plugin but will also work with a slightly modified Spoor plugin.  The placeholder format is: @@property.

Right now properties from config.json and course.json are supported.

Squashed commit messages:
- Added replace task to build.js and dev.js
- Added support for passing both course and config to replace call
- Added 'description' to course.json.  This attribute is useful for the SCORM and xAPI extensions.
- Added shim for missing xAPI property. This is to cover edge cases where the xAPI extension may be defined but its configuration may be incomplete.
- Added a warning mesage for when xAPI activityID has not been set.  The course will not confirm to the spec if this has been left empty.

### CODE REVIEW

**Comment by brian-learningpool** [comment]
This resolves #939.

**Comment by bell87** [comment]
+1

File: grunt/config/replace.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      '_isEnabled': false
+    };
+  } else {
+    // xAPI has been enabled, check if the activityID has been set.
+    if (configJson._xapi.hasOwnProperty('activityID') && configJson._xapi.activityID == '') {
+      grunt.log.writeln('WARNING: xAPI activityID has not been set');
+    }
+  }
+
+  // Combine the course and config JSON so both can be passet to replace.
```
Comment (dennis-learningpool) [inline grunt/config/replace.js]: passet -> passed

**Comment by dennis-learningpool** [comment]
+1

**Comment by moloko** [comment]
+1

File: grunt/config/replace.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  var defaultLanguage = configJson._defaultLanguage || 'en';
+  var courseJson = grunt.file.readJSON(courseDir + defaultLanguage + '/course.json');
+
+  // Backwards compatibility for courses missing 'description'
+  if (!courseJson.hasOwnProperty('description')) {
+    courseJson.description = '';
+  }
+
+  // A shim for edge cases where xAPI has not been configured.
+  if (!configJson.hasOwnProperty('_xapi')) {
```
Comment (ryangadams) [inline grunt/config/replace.js]: I'm somewhat uncomfortable with having hacks for extensions in the core grunt tasks. Do we really need this section?

Comment (oliverfoster) [inline grunt/config/replace.js]: Ditto. What happens when these variables aren't set?

Comment (brian-learningpool) [inline grunt/config/replace.js]: I concur that the shim isn't ideal but it ensures that the XML files don't get created with `@@propertyName` in them.

Comment (ryangadams) [inline grunt/config/replace.js]: Could we add a check for @@{someString}@@ in replaced files and complain if any are found?

Comment (brian-learningpool) [inline grunt/config/replace.js]: Probably, but I think that would have to be handled in a separate task.

Comment (oliverfoster) [inline grunt/config/replace.js]: Templating would have sorted this, string replacement is OK but it ain't got no logic. 
Cool, well, leave it in for the time being then. We can figure it out later. If everyone is OK with that?

Comment (ryangadams) [inline grunt/config/replace.js]: Yes, I'm fine with that...

**Comment by oliverfoster** [comment]
+1

**Pull request merged**